src/components/NcRichText/autolink.js

```diff
@@ -84,20 +109,57 @@ export const parseUrl = (text) => {
 	return text
 }
 
+/**
+ * Try to get path for router link from an absolute or relative URL.
+ *
+ * @param {import('vue-router').default} router - VueRouter instance of the router link
+ * @param {string} url - absolute URL to parse
+ * @return {string|null} a path that can be useed in the router link or null if this URL doesn't match this router config
+ * @example http://cloud.ltd/nextcloud/index.php/app/files/favorites?fileid=2#fragment => /files/favorites?fileid=2#fragment
+ */
 export const getRoute = (router, url) => {
-	// Skip if Router is not defined in app, or baseUrl does not match
-	if (!router || !url.includes(getBaseUrl())) {
+	/**
+	 * http://cloud.ltd /nextcloud /index.php/app/files /favorites?fileid=2#fragment
+	 * |_____origin____|__________router-base__________|_________router-path________|
+	 * |__________base____________|
+	 *                 |___root___|
+	 */
+
+	// Router is not defined in the app => not an app route
+	if (!router) {
 		return null
 	}
 
-	const regexArray = router.getRoutes()
-		// route.regex matches only complete string (^.$), need to remove these characters
-		.map(route => new RegExp(route.regex.source.slice(1, -1), route.regex.flags))
+	const isAbsoluteURL = /^https?:\/\//.test(url)
 
-	for (const regex of regexArray) {
-		const match = url.search(regex)
-		if (match !== -1) {
-			return url.slice(match)
-		}
+	// URL is not a link to this Nextcloud server instance => not an app route
+	if ((isAbsoluteURL && !url.startsWith(getBaseUrl())) || (!isAbsoluteURL && !url.startsWith(getRootUrl()))) {
+		return null
+	}
+
+	// Vue 3: router.options.history.base
+	const routerBase = router.history.base
+
+	const urlWithoutOrigin = isAbsoluteURL ? url.slice(new URL(url).origin.length) : url
+
+	// Remove index.php - it is optional in general case in both, VueRouter base and the URL
+	const urlWithoutOriginAndIndexPhp = url.startsWith((isAbsoluteURL ? getBaseUrl() : getRootUrl()) + '/index.php') ? urlWithoutOrigin.replace('/index.php', '') : urlWithoutOrigin
```

Comment:
Would there be any drawbacks to omit check for isAbsoluteUrl? Apart from finding `index.php` in unexpected place

```suggestion
	const urlWithoutOriginAndIndexPhp = urlWithoutOrigin.replace('/index.php', '')
```